src/lib/checkbox/checkbox.spec.ts

```diff
@@ -735,6 +736,41 @@ describe('MdCheckbox', () => {
     });
   });
 
+  describe('with required ngModel', () => {
+    let checkboxInstance: MdCheckbox;
+    let inputElement: HTMLInputElement;
+    let testComponent: CheckboxWithNgModel;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(CheckboxWithNgModel);
+      fixture.detectChanges();
+
+      let checkboxDebugElement = fixture.debugElement.query(By.directive(MdCheckbox));
+      let checkboxNativeElement = checkboxDebugElement.nativeElement;
+      testComponent = fixture.debugElement.componentInstance;
+      checkboxInstance = checkboxDebugElement.componentInstance;
+      inputElement = <HTMLInputElement>checkboxNativeElement.querySelector('input');
+    });
+
+    it('should validate with RequiredTrue validator', () => {
+      let checkboxElement = fixture.debugElement.query(By.directive(MdCheckbox));
+      let ngModel = checkboxElement.injector.get<NgModel>(NgModel);
+
+      testComponent.isRequired = true;
+      inputElement.click();
+      fixture.detectChanges();
+
+      expect(checkboxInstance.checked).toBe(true);
+      expect(ngModel.valid).toBe(true);
+
+      inputElement.click();
+      fixture.detectChanges();
+
+      expect(checkboxInstance.checked).toBe(false);
+      expect(ngModel.valid).toBe(false);
```

Comment:
Is there a way to check if the error is specifically due to the required validator?